### DISCUSSION TITLE
Consider allowing yaml 1.2.2 next to json/jsonc

### DISCUSSION THREAD

**Discussion by fahhem** [devcontainers/community#29] [Ideas]
Any interest?

Background:
`jsonc`, aka "JSON with comments", while nicer than plain JSON, is still relatively new and doesn't have full ecosystem support yet. It also has some shortcomings shared with JSON, such as not accepting trailing commas in lists/objects (except on a [per-parser exception](https://github.com/microsoft/vscode/issues/102061) basis).

`yaml` can be limited to a JSON equivalent in terms of data structures it supports, and all (or at least most) of the issues with it were allayed in 1.2+ versions of the spec (such as `on` being parsed as `true` when not quoted). It also reduces the need for as many quotes, commas, open and close braces, and first-class supports comments.

**Comment by Chuxel** [comment]
@fahhem Interesting thought! You could raise a proposal at https://github.com/devcontainers/spec for consideration. Thus far, jsonc has been working, but if this is something the community is interested in, it could get added in.

**Reply by fahhem** [reply]
Ah, I thought this was the path for that. Okay, will do!
